openwisp_controller/geo/api/serializers.py

```diff
@@ -38,3 +36,219 @@ class Meta:
         model = Location
         geo_field = 'geometry'
         fields = '__all__'
+
+
+class BaseSerializer(FilterSerializerByOrgManaged, ValidatedModelSerializer):
+    pass
+
+
+class FloorPlanSerializer(BaseSerializer):
+    name = serializers.SerializerMethodField()
+
+    class Meta:
+        model = FloorPlan
+        fields = (
+            'id',
+            'name',
+            'floor',
+            'image',
+            'location',
+            'created',
+            'modified',
+        )
+        read_only_fields = ('created', 'modified')
+
+    def get_name(self, obj):
+        name = '{0} {1} Floor'.format(obj.location.name, ordinal(obj.floor))
+        return name
+
+    def validate(self, data):
+        if data.get('location'):
+            data['organization'] = data.get('location').organization
+        instance = self.instance or self.Meta.model(**data)
+        instance.full_clean()
+        return data
+
+
+class FloorPlanLocationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = FloorPlan
+        fields = (
+            'floor',
+            'image',
+        )
+        extra_kwargs = {'floor': {'required': False}, 'image': {'required': False}}
+
+
+class LocationSerializer(FilterSerializerByOrgManaged, serializers.ModelSerializer):
```

Comment:
Here I haven't use the `gis_serializers.GeoFeatureModelSerializer`, to keep the representation as it is. Using this serializer class the representation is hindered resulting in not loading the appropriate field in HTML form fields in the DRF Browsable Page, Shiuld I add it?, or it is ok.